client/dashboard/task-list/index.js

```diff
@@ -29,30 +34,40 @@ const getTasks = () => {
 			onClick: noop,
 		},
 		{
+			key: 'products',
 			title: __( 'Add your first product', 'woocommerce-admin' ),
 			description: __(
 				'Add products manually, import from a sheet or migrate from another platform',
 				'wooocommerce-admin'
 			),
-			before: <i class="material-icons-outlined">add_box</i>,
+			before: tasks.products ? (
+				<i class="material-icons-outlined">check_circle</i>
+			) : (
+				<i class="material-icons-outlined">add_box</i>
+			),
 			after: <i class="material-icons-outlined">chevron_right</i>,
-			onClick: noop,
+			onClick: () => updateQueryString( { task: 'products' } ),
```

Comment:
Guessing we will probably want to layer in some tracks on this click event too.

Reply:
Good point.  I'll probably handle this in a follow-up issue if that's okay.  I'm anticipating a small refactor to the list click-handlers to DRY this up a bit, but need to see all of the tasks here to know if that will work.

Issue for tracks outlined [here](https://github.com/woocommerce/woocommerce-admin/issues/2671)